data/trie/trieStorageManager.go

```diff
@@ -484,7 +484,9 @@ func (tsm *trieStorageManager) disconnectSnapshot() {
 		return
 	}
 
-	disconnectSnapshot(snapshot)
+	if err := disconnectSnapshot(snapshot); err != nil {
```

Comment:
I think it will look better if you split it like this:

err := disconnectSnapshot(s.DBWriteCacher)
if err != nil .....
 
Also, we use the same pattern throughout the rest of the project.

Reply:
Done!